drivers/FileHandle.h

```diff
@@ -25,62 +25,85 @@ namespace mbed {
  */
 class FileHandle {
-
 public:
-    MBED_DEPRECATED_SINCE("mbed-os-5.4",
-        "The mbed 2 filesystem classes have been superseeded by the FileSystem api, "
-        "Replaced by File")
-    FileHandle() {}
+    virtual ~FileHandle() {}
 
-    /** Write the contents of a buffer to the file
+    /** Read the contents of a file into a buffer
      *
-     *  @param buffer the buffer to write from
-     *  @param length the number of characters to write
+     *  @param buffer   The buffer to read in to
+     *  @param size     The number of bytes to read
+     *  @return         The number of bytes read, 0 at end of file, negative error on failure
+     */
+    virtual ssize_t read(void *buffer, size_t len) = 0;
+
+    /** Write the contents of a buffer to a file
      *
-     *  @returns
-     *  The number of characters written (possibly 0) on success, -1 on error.
+     *  @param buffer   The buffer to write from
+     *  @param size     The number of bytes to write 
+     *  @return         The number of bytes written, negative error on failure
      */
-    virtual ssize_t write(const void* buffer, size_t length) = 0;
+    virtual ssize_t write(const void *buffer, size_t len) = 0;
 
-    /** Close the file
+    /** Close a file
      *
-     *  @returns
-     *  Zero on success, -1 on error.
+     *  @return         0 on success, negative error code on failure
      */
     virtual int close() = 0;
 
-    /** Function read
-     *  Reads the contents of the file into a buffer
+    /** Flush any buffers associated with the file
      *
-     *  @param buffer the buffer to read in to
-     *  @param length the number of characters to read
-     *
-     *  @returns
-     *  The number of characters read (zero at end of file) on success, -1 on error.
+     *  @return         0 on success, negative error code on failure
      */
-    virtual ssize_t read(void* buffer, size_t length) = 0;
+    virtual int sync() = 0;
 
-    /** Check if the handle is for a interactive terminal device.
-     * If so, line buffered behaviour is used by default
+    /** Check if the file in an interactive terminal device
      *
-     *  @returns
-     *    1 if it is a terminal,
-     *    0 otherwise
+     *  @return         True if the file is a terminal
      */
     virtual int isatty() = 0;
 
+    /** Move the file position to a given offset from from a given location
+     *
+     *  @param offset   The offset from whence to move to
+     *  @param whence   The start of where to seek
+     *      SEEK_SET to start from beginning of file,
+     *      SEEK_CUR to start from current position in file,
+     *      SEEK_END to start from end of file
+     *  @return         The new offset of the file
+     */
+    virtual off_t seek(off_t offset, int whence = SEEK_SET) = 0;
+
+    /** Get the file position of the file
+     *
+     *  @return         The current offset in the file
+     */
+    virtual off_t tell() = 0;
```

Comment:
Not dug into that one in detail, but it seems to me the tell implementation might be too optimised, to the point of being faulty.

Unlike seek, it doesn't check handle validity, it doesn't return sticky errors. Shouldn't it?

But yes, ftell[o] is standard POSIX, so why not. (I just failed to spot it)